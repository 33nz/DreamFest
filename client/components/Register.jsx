import React, { useState } from 'react'
import { register, isAuthenticated } from 'authenticare/client'
// import { createUser } from "../../server/db/connection"

export default function Register(props) {
  const [username, setUsername] = useState("")
  const [password, setPassword] = useState("")
  const [garden, setGarden] = useState(Number(""))
  const baseUrl = "/api/v1"

  const handleClick = () => {
    register(
      {
        username: username,
        password: password,
        garden: Number(garden),
      },
      { baseUrl }
    )
      .then((token) => {
        console.log(token)
        if (isAuthenticated()) {
          return props.history.push('/garden')
        }
        return null
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }

  return (
    <div className="registerContainer container">
      <div className="leftRegister">
        <div className="registerTitle">
          <h1>Register</h1>
        </div>
        <label className="label">Username</label>
        <input
          className="input"
          type="username"
          value={username}
          placeholder="username"
          onChange={(e) => setUsername(e.target.value)}
        ></input>
        <label className="label">Password</label>
        <input
          className="input"
          type="password"
          value={password}
          placeholder="password"
          onChange={(e) => setPassword(e.target.value)}
        ></input>
        <label className="label">My Garden</label>
        <select
          onChange={(e) => setGarden(e.target.value)}
          className="select"
          name="garden"
          id="name"
        >
          <option hidden>Select from this list</option>
          <option value="1">Kelmarna Gardens</option>
          <option value="2">Kingsland Community Orchard</option>
          <option value="3">Devonport Community Garden</option>
        </select>
        <button
          type="button"
          className="button is-primary"
          onClick={handleClick}
          data-testid="submitButton"
        >
          Register
        </button>
      </div>
      <div className="rightRegister">
        <img
          src="./images/comGardenPlant.png"
          alt=""
          style={{ width: '600px', height: '500px' }}
        />
      </div>
    </div>
  )
}
