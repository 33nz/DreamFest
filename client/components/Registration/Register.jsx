import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { registerUser } from './registerHelper'
import { useAuth0 } from '@auth0/auth0-react'

export function Register () {
  const authUser = useAuth0().user
  const auth0Id = authUser.sub
  console.log(authUser)

  const [form, setForm] = useState({
    firstName: '',
    lastName: '',
    username: '',
    gardenId: null,
    email: authUser.email,
    auth0Id: auth0Id
  })
  const history = useHistory()

  function handleChange (e) {
    const { name, value } = e.target
    setForm({
      ...form,
      [name]: value
    })
  }

  function handleClick (e) {
    e.preventDefault()
    registerUser(form, history.push)
  }

  return (
    <>
      <form className='column'>
        <div className="field">
          <label htmlFor='firstName' className='label'>First Name</label>
          <input
            className='input'
            id='firstName'
            name='firstName'
            value={form.firstName}
            placeholder='First Name'
            onChange={handleChange}
          ></input>
        </div>
        <div className="field">
          <label htmlFor='lastName' className='label'>Last Name</label>
          <input
            className='input'
            id='lastName'
            name='lastName'
            value={form.lastName}
            placeholder='Last Name'
            onChange={handleChange}
          ></input>
        </div>
        <div className="field">
          <label htmlFor='username' className='label'>Username</label>
          <input
            className='input'
            id='username'
            name='username'
            value={form.username}
            placeholder='Username'
            onChange={handleChange}
          ></input>
        </div>
        <div className="field">
          <label htmlFor='garden' className='label'>My Garden</label>
          <select
            onChange={handleChange}
            className='select'
            name='gardenId'
            id='garden'
          >
            <option hidden>Select from this list</option>
            <option value={1}>Kelmarna Gardens</option>
            <option value={2}>Kingsland Community Orchard</option>
            <option value={3}>Devonport Community Garden</option>
          </select>
        </div>
        <button
          type='button'
          className='button'
          onClick={handleClick}
          data-testid='submitButton'
        >
            Register
        </button>
      </form>
      <div className='column'>
        <img src='./images/comGardenPlant.png' alt='Person gardening with trowel' />
      </div>
    </>
  )
}
