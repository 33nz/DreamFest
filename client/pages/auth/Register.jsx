import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'

import { registerUser } from './registerHelper'

export default function Register (props) {
  const [form, setForm] = useState({
    username: '',
    password: '',
    gardenId: null,
    email: ''
  })
  const history = useHistory()

  function handleChange (e) {
    const { name, value } = e.target
    setForm({
      ...form,
      [name]: value
    })
  }

  function handleClick (e) {
    e.preventDefault()
    registerUser(form, history.push)
  }

  return (
    <>
      <div className="column ">
        <form className="inputForm">
          <label htmlFor="username" className="label">Username</label>
          <input
            className="input"
            id="username"
            name="username"
            value={form.username}
            placeholder="Username"
            onChange={handleChange}
          ></input>
          <label htmlFor="password" className="label">Password</label>
          <input
            className="input"
            id="password"
            type="password"
            name="password"
            value={form.password}
            placeholder="Password"
            onChange={handleChange}
          ></input>

          <label htmlFor="email" className="label">Email</label>
          <input
            className="input"
            id="email"
            type="email"
            name="email"
            value={form.email}
            placeholder="Email"
            onChange={handleChange}
          ></input>

          <label htmlFor="garden" className="label">My Garden</label>
          <select
            onChange={handleChange}
            className="select"
            name="gardenId"
            id="garden"
          >
            <option hidden>Select from this list</option>
            <option value={1}>Kelmarna Gardens</option>
            <option value={2}>Kingsland Community Orchard</option>
            <option value={3}>Devonport Community Garden</option>
          </select>
          <button
            type="button"
            className="button ml-3 registerButton"
            onClick={handleClick}
            data-testid="submitButton"
          >
            Register
          </button>
        </form>
      </div>
      <div className="column ">
        <div className="imageFit">
          <img
            src="./images/comGardenPlant.png"
            alt=""
          />
        </div>
      </div>
    </>
  )
}
