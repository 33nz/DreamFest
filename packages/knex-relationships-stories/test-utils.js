const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

const render = (response) => within(new JSDOM(response.text).window.document)

module.exports = { render }
