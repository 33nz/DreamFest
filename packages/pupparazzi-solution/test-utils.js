const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

export const render = (response) =>
  within(new JSDOM(response.text).window.document)
