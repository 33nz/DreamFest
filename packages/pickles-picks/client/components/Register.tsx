import React, {
  useEffect,
  useState,
  ChangeEvent,
  MouseEvent,
  FormEvent,
  BaseSyntheticEvent,
  FormEventHandler,
  DetailedHTMLProps,
  FormHTMLAttributes,
} from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useNavigate } from 'react-router-dom'
import { Store } from '../../types'

import { authError, registerUserRequest } from '../actions/auth'

function Register() {
  const navigateTo = useNavigate()
  const dispatch = useDispatch()
  const auth = useSelector((redux: Store) => redux.auth)

  const [formData, setFormData] = useState({
    username: '',
    email_address: '',
    password: '',
    confirm_password: '',
  })

  useEffect(() => {
    dispatch(authError(''))
  }, [])

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setFormData((prevFormData) => {
      return {
        ...prevFormData,
      }
    })
  }

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()
   e.currentTarget.reset()

    const { password, confirm_password, username, email_address } = formData

    if (confirm_password != password) {
      dispatch(authError("Passwords don't match"))
    } else {
      const confirmSuccess = () => navigateTo('/')
      const userInfo = { username, password, email_address }
      dispatch(registerUserRequest(userInfo, confirmSuccess))
    }
  }

  return (
    <form className="Register form box" onSubmit={(e)=> console.log(e)}>
      <h1 className="title is-2">Register</h1>
      <hr />
      {auth.errorMessage && (
        <span className="has-text-danger is-large">{auth.errorMessage}</span>
      )}
      <label className="column is-6 is-offset-one-quarter label is-large has-text-centered">
        Username
        <input
          required
          className="input is-large has-text-centered is-fullwidth"
          placeholder="User Name"
          type="text"
          name="username"
          autoComplete="username"
          onChange={handleChange}
          value={formData.username}
        />
      </label>
      <div className="columns">
        <label className="column is-6 label is-large has-text-centered">
          Email Address
          <input
            required
            className="input is-large has-text-centered is-fullwidth"
            placeholder="Email Adress"
            type="text"
            name="email_address"
            onChange={handleChange}
            value={formData.email_address}
          />
        </label>
      </div>
      <br />
      <div className="columns">
        <label className="column is-6 label is-large has-text-centered">
          Password
          <input
            required
            className="input is-large has-text-centered is-fullwidth"
            placeholder="Password"
            type="password"
            name="password"
            autoComplete="new-password"
            onChange={handleChange}
            value={formData.password}
          />
        </label>
        <label className="column is-6 label is-large has-text-centered">
          Confirm Password
          <input
            required
            className="input is-large has-text-centered is-fullwidth"
            placeholder="Confirm Password"
            type="password"
            name="confirm_password"
            autoComplete="new-password"
            onChange={handleChange}
            value={formData.confirm_password}
          />
        </label>
      </div>
      <input
        className="button is-success is-large is-fullwidth"
        value="Register"
        type="submit"
      />
    </form>
  )
}

export default Register
