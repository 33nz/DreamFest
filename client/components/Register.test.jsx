import React from 'react'
import { fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import Register from './Register.jsx'
import { renderWithRedux } from '../test-utils'

describe('Username Input Field Updates', () => {
  it('gets the text', () => {
    const { queryByPlaceholderText } = renderWithRedux(<Register />)
    const usernameInput = queryByPlaceholderText('username')
    fireEvent.change(usernameInput, { target: { value: 'username' } })
    expect(usernameInput.value).toBe('username')
  })
})

describe('Password Input Field Updates', () => {
  it('gets the text', () => {
    const { queryByPlaceholderText } = renderWithRedux(<Register />)
    const passwordInput = queryByPlaceholderText('username')
    fireEvent.change(passwordInput, { target: { value: 'password' } })
    expect(passwordInput.value).toBe('password')
  })
})
// Don: This test isn't useful enough - let's chat
describe('Submit Button Finds Handleclick', () => {
  // remove the next line after fixing the test
  // eslint-disable-next-line jest/expect-expect
  it('sends the submit button', () => {
    const { queryByTestId } = renderWithRedux(<Register />)
    fireEvent.click(queryByTestId('submitButton'))
  })
})
