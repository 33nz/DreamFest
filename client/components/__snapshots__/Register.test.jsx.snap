// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Renders renders correctly 1`] = `
<div
  className="registerContainer container"
>
  <div
    className="leftRegister"
  >
    <div
      className="registerTitle"
    >
      <h1>
        Register
      </h1>
    </div>
    <label
      className="label"
    >
      Username
    </label>
    <input
      className="input"
      onChange={[Function]}
      placeholder="username"
      type="username"
      value=""
    />
    <label
      className="label"
    >
      Password
    </label>
    <input
      className="input"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <label
      className="label"
    >
      My Garden
    </label>
    <select
      className="select"
      id="name"
      name="garden"
      onChange={[Function]}
    >
      <option
        hidden={true}
      >
        Select from this list
      </option>
      <option
        value="1"
      >
        Kelmarna Gardens
      </option>
      <option
        value="2"
      >
        Kingsland Community Orchard
      </option>
      <option
        value="3"
      >
        Devonport Community Garden
      </option>
    </select>
    <button
      className="button is-primary"
      data-testid="submitButton"
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
  </div>
  <div
    className="rightRegister"
  >
    <img
      alt=""
      src="./images/comGardenPlant.png"
      style={
        Object {
          "height": "500px",
          "width": "600px",
        }
      }
    />
  </div>
</div>
`;
